components/editor/modules/figure/caption.js

```diff
@@ -82,20 +82,31 @@ const getSerializer = options => {
       byline
     ] = object.nodes
 
+    const children = [
+      ...inlineSerializer.toMdast(
+        caption.nodes,
+        0,
+        object,
+        rest
+      )
+    ]
+    const bylineChildren = bylineModule.helpers.serializer.toMdast(
+        byline.nodes,
+        0,
+        object,
+        rest
+    )
+
+    if (bylineChildren.length && bylineChildren[0].value !== '') {
```

Comment:
`(bylineChildren.length && !(bylineChildren.length === 1 && bylineChildren[0].value === '')` would be safer otherwise if there is a empty first text node, for some odd reason, it will skip the byline. Or use https://github.com/syntax-tree/mdast-util-to-string and do `toString(bylineChildren).trim()`